packages/pwa-kit-create-app/assets/bootstrap/js/overrides/app/utils/translations.js

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * Dynamically import the translations/messages for a given locale
+ * @param {string} locale
+ * @returns {Promise<Object>} The messages (compiled in AST format) in the given locale.
+ *      If the translation file is not found, return an empty object (so react-intl would fall back to the inline messages)
+ */
+export const fetchTranslations = async (locale) => {
+    const targetLocale =
+        typeof window === 'undefined'
+            ? process.env.USE_PSEUDOLOCALE === 'true'
+                ? 'en-XA'
+                : locale
+            : locale
+
+    let module
+    try {
+        module = await import(`../../../translations/compiled/${targetLocale}.json`)
```

Comment:
Our goal should be that the translation feature just works magically without leaking implementation detail in the generated project.

Reply:
If the generated project has a folder structure like this

```
my-project/
├── overrides/
│   └── app/
│       └── utils/
│           └── translations.js
└── translations/
    └── compiled.json
```

Then I believe you have too many `..`? Shouldn't it be `../../translations/compiled/${targetLocale}.json` ？

Reply:
alternatively, have we considered moving the compiled json files into the static folder. that way you could do


```suggestion
import { getAssetUrl } from '@salesforce/pwa-kit-react-sdk/ssr/universal/utils'

        module = await import(getAssetUrl(`${targetLocale}.json`))
```

Reply:
Yeah. Generator app has the overrides folder which slightly changes the relative path so I had to copy the file over to the generated app.

Reply:
Any suggestions for how to approach this differently @kevinxh ?

This might be a thing that we monitor via e2e sanity tests that make sure localization works on multisite in e2e nightlies. 

We do need a fix (even if this one is suboptimal) to unblock v3 release... and without a better understanding of the problem it's hard for me to propose one, perhaps we should huddle and discuss?

Reply:
did our previous fix not working? https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1215/files

I would suggest that adding the translation files to the bundle and fetch translations just like fetching any bundle assets like favicon.